src/Workspaces/VisualBasic/Portable/Simplification/AbstractVisualBasicSimplifier.AbstractExpressionRewriter.vb

```diff
@@ -122,7 +139,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Simplification
                 Return newToken
             End Function
 
-            Protected Function SimplifyExpression(Of TExpression As ExpressionSyntax)(
+            Protected Function SimplifyExpression(Of TExpression As SyntaxNode)(
```

Comment:
See corresponding C# implementation of [SimplifyExpression](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Simplification/AbstractCSharpReducer.AbstractExpressionRewriter.cs,116). Notice that `TExpression` is constrained to `SyntaxNode`.

Specifically, the new Reducer (`VisualBasicInferredMemberNameReducer.Rewriter`) needs to simplify a `SimpleArgumentSyntax` (for tuples) and a `NamedFieldInitializerSyntax` (for anonymous types). Copied relevant code below:

```VB
     Public Overrides Function VisitSimpleArgument(node As SimpleArgumentSyntax) As SyntaxNode
                CancellationToken.ThrowIfCancellationRequested()

                Dim newNode = MyBase.VisitSimpleArgument(node)

                If node.IsParentKind(SyntaxKind.TupleExpression) Then
                    Return SimplifyExpression(
                        node,
                        newNode:=newNode,
                        simplifier:=AddressOf SimplifyTupleName)
                End If

                Return newNode
            End Function

            Public Overrides Function VisitNamedFieldInitializer(node As NamedFieldInitializerSyntax) As SyntaxNode
                Dim newNode = MyBase.VisitNamedFieldInitializer(node)

                Return SimplifyExpression(
                    node,
                    newNode:=newNode,
                    simplifier:=AddressOf SimplifyNamedFieldInitializer)
            End Function

        End Class
```